src/group_theory/group_action/sub_mul_action.lean

```diff
@@ -144,7 +145,36 @@ instance : mul_action R p := p.mul_action'
 
 end
 
-end mul_action
+
+/-- Orbits in a `sub_mul_action` coincide with orbits in the ambient space. -/
+lemma orbit_of_sub_mul {p : sub_mul_action R M} (m : p) :
+  (mul_action.orbit R m : set M) = mul_action.orbit R (m : M) := rfl
```

Comment:
This lemma is a tautology, both sides are syntactically identical. Did you mean
```lean
lemma orbit_of_sub_mul {p : sub_mul_action R M} (m : p) :
  coe '' (mul_action.orbit R m) = mul_action.orbit R (m : M) := sorry
```

Reply:
Actually, I meant what I had written, because at some point I felt it had to exist.
(It is not clear to me why the orbit, a member of (set p), is automatically viewed as a member of (set M).
Let me try to understand what you wrote.

Reply:
(I switched my lemma to yours, but it is not clear to me why the first one in syntactically obvious and not yours…)

Reply:
> (It is not clear to me why the orbit, a member of (set p), is automatically viewed as a member of (set M).

That's precisely the problem, that's not what you asked lean to do. When you write `(mul_action.orbit R m : set M)`, that means "`mul_action.orbit R m` will be a term of type `set M`", from which the elaborator concludes "`m` must be a term of type `M`", and replaces it with `↑m`".

My guess is that you thought you'd written
```lean
(↑(mul_action.orbit R m) : set M)
```
Which doesn't work because no such automatic coercion exists

Reply:
> it is not clear to me why the first one in syntactically obvious and not yours

Put your cursor after the `:=` in your original lemma. The goal view will show that both sides are already exactly the same.

Reply:
You taught me something about how Lean's elaborator “thinks”. 
I really expected that seeing `(mul_action.orbit R m : set M)', it would want to make `(mul_action.orbit R m)' a member of `set M'.

Reply:
Actually, I wrote this lemma by a mere desire of completing the square (If something is proven about stabilizer, something about orbits has to, and the lemma for orbits was the wrong one, as you observed!), but I don't think I had to use it.